georocket-client-api/src/main/java/io/georocket/client/StoreClient.java

```diff
@@ -657,13 +657,46 @@ private void update(UpdateAction action, UpdateTarget target, String query, Stri
       return;
     }
 
-    JsonObject body = new JsonObject()
-      .put("action", action.toString())
-      .put("target", target.toString())
-      .put("updates", new JsonArray(updates));
+    HttpMethod method;
+
+    switch (action) {
```

Comment:
Now that we have different EndPoints we can put this code in new classes (`PropertiesClient` and `TagsClient`). I think this should be a separate pull request!